Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -464,11 +471,80 @@ public CsharpCompiler(ErrorReporter reporter)
 
       CompileDatatypeDestructorsAndAddToInterface(dt, wr, interfaceTree, DtT_TypeArgs);
 
+      DtDowncastClone(dt, interfaceTree, nonGhostTypeArgs, @interface: true);
```

Comment:
I would suggest to avoid the '@' when you can have more descriptive names that don't use keywords, e.g. `isInterface` or `hasInterface` or `containsInterface`. '@' were invented for compatibility with other languages, not really to allow the use of keywords as variables (except in rare cases)